addons/hr_fleet/models/fleet_vehicle.py

```diff
@@ -19,12 +30,158 @@ def _compute_mobility_card(self):
+        #       we remove it's current driver_id
+        if 'fleet_id' in vals:
+            if not is_internal:
+                vals['driver_employee_id'] = False
+                vals['future_driver_employee_id'] = False
+            else:
+                # Prefetch all employees we will need
+                employees = self.env['hr.employee'].search([
+                    ('address_home_id', 'in', (self.driver_id | self.future_driver_id).ids),
+                ])
+                for vehicle in self:
+                    if 'driver_employee_id' not in vals:
+                        driver_id = vals['driver_id'] if 'driver_id' in vals else vehicle.driver_id.id
+                        driver_employee_id = employees.filtered(lambda e: e.address_home_id.id == driver_id)
+                        if len(driver_employee_id) == 1:
+                            vehicle.driver_employee_id = driver_employee_id
+                        else:
+                            vehicle.write({
+                                'driver_id': False,
+                                'driver_employee_id': False,
+                            })
+                    if 'future_driver_employee_id' not in vals:
+                        future_driver_id = vals['future_driver_id'] if 'future_driver_id' in vals else \
+                            vehicle.future_driver_id.id
+                        future_driver_employee_id = employees.filtered(lambda e: (
+                            e.address_home_id.id == future_driver_id
+                        ))
+                        if len(future_driver_employee_id) == 1:
+                            vehicle.future_driver_employee_id = future_driver_employee_id
+                        else:
+                            vehicle.write({
+                                'future_driver_id': False,
+                                'future_driver_employee_id': False,
+                            })
+
+    @api.model
+    def create(self, vals):
+        self._update_create_write_vals(vals)
+        return super().create(vals)
 
     def write(self, vals):
-        res = super().write(vals)
-        if 'mobility_card' in vals:
-            vehicles = self.env['fleet.vehicle'].search([('driver_id', 'in', (self.user_id.partner_id | self.sudo().address_home_id).ids)])
-            vehicles._compute_mobility_card()
+        self._update_create_write_vals(vals)
+        return super().write(vals)
+
+class FleetVehicleOdometer(models.Model):
+    _inherit = 'fleet.vehicle.odometer'
+
+    driver_employee_id = fields.Many2one(
+        related='vehicle_id.driver_employee_id', string='Current Driver (Employee)',
+        readonly=True,
+    )
+
+class FleetVehicleAssignationLog(models.Model):
+    _inherit = 'fleet.vehicle.assignation.log'
+
+    driver_employee_id = fields.Many2one('hr.employee', 'Driver (Employee)')
+    fleet_is_internal = fields.Boolean(related='vehicle_id.fleet_is_internal')
```

Comment:
Why not add those 2 fields in the existing fleet_vehicle_assignation_log file that you deleted?

Reply:
Mainly to have the same file structure as the parent models, since fleet.vehicle.assignation.log is part of fleet/models/fleet_vehicle.py, it's easier to work with but not really important.

I don't mind readding the file and adding those two fields to it, what is preferred?

Reply:
It's better to have one file per model. That way it's easier to find it when you're looking for it. Readding the file and putting the two fields in it would be better